core/src/main/scala/pureconfig/error/ConfigReaderFailures.scala

```diff
@@ -0,0 +1,24 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package pureconfig.error
+
+/**
+ * A non-empty list of ConfigReader failures
+ */
+case class ConfigReaderFailures(head: ConfigReaderFailure, tail: List[ConfigReaderFailure]) {
+
+  def toList: List[ConfigReaderFailure] = head +: tail
+
+  def +(failure: ConfigReaderFailure): ConfigReaderFailures =
```

Comment:
Suggestion: I would go with `+:` for uniformity with the standard `List`, preserve the visual ordering semantics that of having the new head on the left, and to avoid the weirdness around string concatenation.
Feel free to disagree.